includes/class-sensei-course.php

```diff
@@ -2005,6 +2005,37 @@ public function get_progress_statement( $course_id, $user_id ) {
 
 	}
 
+	/**
+	 * Returns a summary of course progress statistics in keyed array.
+	 *
+	 * @param int $course_id The id of the course.
+	 *
+	 * @return array $stats An array of progress stats.
+	 *               $stats['lessons_count'] The total number of lessons in the course.
+	 *               $stats['completed_lessons_count'] The total number of completed lessons of the course by the user.
+	 *               $stats['completed_lessons_percentage'] The completed lessons percentage relative to total number of lessons.
+	 */
+	public function get_progress_stats( int $course_id ): array {
+		$lessons_count           = count( \Sensei()->course->course_lessons( $course_id, null, 'ids' ) );
+		$completed_lessons_count = count( \Sensei()->course->get_completed_lesson_ids( $course_id ) );
+		$stats                   = [
+			'lessons_count'                => $lessons_count,
+			'completed_lessons_count'      => $completed_lessons_count,
+			'completed_lessons_percentage' => \Sensei_Utils::quotient_as_absolute_rounded_percentage( $completed_lessons_count, $lessons_count, 2 ),
+		];
+
+		/**
+		 * Filter the course progress stats.
+		 *
+		 * @since 3.14.4
+		 * @param array $stat An array of course progress stats.
+		 *              $stats['lessons_count'] The total number of lessons in the course.
+		 *              $stats['completed_lessons_count'] The total number of completed lessons of the course by the user.
+		 *              $stats['completed_lessons_percentage'] The completed lessons percentage relative to total number of lessons.
+		 */
+		return apply_filters( 'sensei_course_progress_stats', $stats );
```

Comment:
If we add a filter for that, I think we could inform something else to give a context of the information. Maybe the `$course_id`?

Reply:
🤦 Of course! That makes a lot of sense. 😄 I updated it and now it receives the $course_id as a second parameter.